repos/system_upgrade/el7toel8/actors/migratentp/libraries/ntp2chrony.py

```diff
@@ -0,0 +1,664 @@
+#!/usr/bin/python
+#
+# Convert ntp configuration to chrony
+#
+# Copyright (C) 2018  Miroslav Lichvar <mlichvar@redhat.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import argparse
+import ipaddress
+import os
+import os.path
+import re
+import subprocess
+import sys
+
+# python2 compatibility hacks
+if sys.version_info[0] < 3:
+    from io import open
+    reload(sys)
+    sys.setdefaultencoding("utf-8")
+
+class NtpConfiguration(object):
+    def __init__(self, root_dir, ntp_conf, step_tickers, verbose):
+        self.root_dir = root_dir if root_dir != "/" else ""
+        self.ntp_conf_path = ntp_conf
+        self.step_tickers_path = step_tickers
+        self.verbose = verbose
+
+        self.enabled_services = set()
+        self.step_tickers = []
+        self.time_sources = []
+        self.fudges = {}
+        self.restrictions = {
+                # Built-in defaults
+                ipaddress.ip_network(u"0.0.0.0/0"): set(),
+                ipaddress.ip_network(u"::/0"): set(),
+        }
+        self.keyfile = ""
+        self.keys = []
+        self.trusted_keys = []
+        self.driftfile = ""
+        self.statistics = []
+        self.leapfile = ""
+        self.tos_options = []
+        self.ignored_directives = set()
+        self.ignored_lines = []
+
+        #self.detect_enabled_services()
+        self.parse_step_tickers()
+        self.parse_ntp_conf()
+
+    def detect_enabled_services(self):
+        for service in ["ntpdate", "ntpd", "ntp-wait"]:
+            if os.path.islink("{}/etc/systemd/system/multi-user.target.wants/{}.service"
+                    .format(self.root_dir, service)):
+                self.enabled_services.add(service)
+        if self.verbose > 0:
+            print("Enabled services found in /etc/systemd/system: " +
```

Comment:
please you string formatting

Reply:
In this case, we are talking about script that is used probably somewhere else as well and tested already. I would keep it as it is as it can follow even different practices in other projects. I think this is something what we do not have mentioned anywhere even best practices - how we will look at the third party code. Of course, we can keep it as it is code going just into the project, but... I feel that we should think that yet.

Reply:
Should be fixed in HEAD.